quickstart-existing-stack-benchmark/run_example.md

```diff
@@ -0,0 +1,237 @@
+
+
+## Benchmarking Steps
+
+
+### 1. Prepare workload specification (profile)
+
+You will need a `yaml` specification to tell `inference-perf` how to generate the _Workload_ that would be used to benchmark your stack. `inference-perf` will generate prompts (AKA a _Data Set_) with timing (AKA _Load_).
+
+Several workload examples are available under `llm-d-benchmark/workload/profiles/inference-perf`. We demonstrate with `shared_prefix_synthetic`.
+
+<details>
+<summary>Click to view `shared_prefix_synthetic.yaml.in`</summary>
+
+For example, `shared_prefix_synthetic.yaml.in`:
+```yaml
+load:
+  type: constant
+  stages:
+  - rate: 2
+    duration: 50
+  - rate: 5
+    duration: 50
+  - rate: 8
+    duration: 50
+  - rate: 10
+    duration: 50
+  - rate: 12
+    duration: 50
+  - rate: 15
+    duration: 50
+  - rate: 20
+    duration: 50
+api:
+  type: completion
+  streaming: true
+server:
+  type: vllm
+  model_name: REPLACE_ENV_LLMDBENCH_DEPLOY_CURRENT_MODEL
+  base_url: REPLACE_ENV_LLMDBENCH_HARNESS_STACK_ENDPOINT_URL
+  ignore_eos: true
+tokenizer:
+  pretrained_model_name_or_path: REPLACE_ENV_LLMDBENCH_DEPLOY_CURRENT_TOKENIZER
+data:
+  type: shared_prefix
+  shared_prefix:
+    num_groups: 32                # Number of distinct shared prefixes
+    num_prompts_per_group: 32     # Number of unique questions per shared prefix
+    system_prompt_len: 2048       # Length of the shared prefix (in tokens)
+    question_len: 256             # Length of the unique question part (in tokens)
+    output_len: 256               # Target length for the model's generated output (in tokens)
+report:
+  request_lifecycle:
+    summary: true
+    per_stage: true
+    per_request: true
+storage:
+  local_storage:
+    path: /workspace
+```
+</details>
+
+If you want to create your own `inference-perf` profile then save your custom `yaml` specification with a `.yaml.in` suffix in the same directory.
+
+⚠️ Unless you know exactly what you are doing, you should only edit the `load` and `data` sections. `load` tell `inference-perf` how many sub-test ("_stages_") to run; each stage has a desired QPS and duration. `data` defines how to create the Data Set; the configuration parameters are different for each `type`.
+
+
+### 2. Log on into your cluster
+Run `oc login ...`
+
+
+### 3. Gather required parameters (mostly information about your `llm-d` stack)
+
+* **Namespace**:
+  The K8S namespace / RHOS project being use.
+  <details>
+  <summary>Click to view bash code snippet</summary>
+
+  ```bash
+  oc config current-context | awk -F / '{print $1}'
+  ```
+  </details>
+
+* **Hugging-Face Token**: 
+  Can obtain the one used in exiting `llm-d` stack.
+  <details>
+  <summary>Click to view bash code snippet</summary>
+
+  ```bash
+  oc get secrets llm-d-hf-token -o jsonpath='{.data.*}' | base64 -d
+  ```
+  </details>
+
+* **Model**: 
```

Comment:
If `-t` is **not** `modelservice` and **not** `standalone` , then the model is set to `auto` and automatically discovered.

Reply:
Great. I will remove it.